### PR TITLE
Enable `/warnAsError`

### PR DESCRIPTION
**Pull request by dougbu**
- Allow some warnings related to closed issues

### CODE REVIEW

File: src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj [hunk 1]
```diff
@@ -13,8 +13,6 @@
     <NoWarn>$(NoWarn);NU5100</NoWarn>
     <!-- It is intentional to not include build/netstandard2.0/Microsoft.NET.Sdk.Razor.targets -->
     <NoWarn>$(NoWarn);NU5129</NoWarn>
-    <!-- Ignore file copy warnings -->
-    <NoWarn>$(NoWarn);MSB3030;MSB4181</NoWarn>
```
Comment (dougbu) [inline src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj:17]: @pranavkm and @JunTaoLuo these warnings should be behind us, correct❔

File: Directory.Build.props [hunk 2]
```diff
@@ -101,14 +101,11 @@
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <!-- Don't make missing XML docs a fatal build error, but still surface so we have visibility into undocumented APIs. -->
     <WarningsNotAsErrors>$(WarningsNotAsErrors);CS1591</WarningsNotAsErrors>
```
Comment (dougbu) [inline Directory.Build.props:103]: @Pilchie are we tracking the many places we turn CS1591 off completely (with `$(NoWarn)` settings)❔
And, do we intend to fill those gaps, where they exist in product, non-analyzer, non-`msbuild` task, non-tool code❔

**Comment by dougbu** [comment]
/fyi will run an internal pipeline once we drive warnings down enough for PR validation to succeed. We have a small amount of site extensions code that doesn't build in PR validation.

**Comment by dougbu** [comment]
As expected, #22992 blocks this. Warnings mentioned there are visible in all test builds.

File: build.ps1 [hunk 3]
```diff
@@ -358,17 +358,12 @@ if (-not $foundJdk -and $RunBuild -and ($All -or $BuildJava) -and -not $NoBuildJ
...
-# Though VS Code may indicate $nodeReuse, $warnAsError and $msbuildEngine are unused, tools.ps1 uses them.
+# Though VS Code may indicate $nodeReuse and $msbuildEngine are unused, tools.ps1 uses them.
 
 # Disable node reuse - Workaround perpetual issues in node reuse and custom task assemblies
 $nodeReuse = $false
 $env:MSBUILDDISABLENODEREUSE=1
 
-# Our build often has warnings that we can't fix, like "MSB3026: Could not copy" due to race
-# conditions in building C++
-# Fixing this is tracked by https://github.com/dotnet/aspnetcore-internal/issues/601
```
Comment (dougbu) [inline build.ps1:369]: Reminder to self: Close dotnet/aspnetcore#601 when this PR goes in 😺

File: src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj (hunk 1 above)
Comment (dougbu) [reply src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj:17]: /ping @pranavkm @JunTaoLuo

File: Directory.Build.props (hunk 2 above)
Comment (dougbu) [reply Directory.Build.props:103]: /ping @Pilchie

File: src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj (hunk 1 above)
Comment (pranavkm) [reply src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj:17]: The first one should be fixed. I'm not sure what the second warning is for.:

> Emit MSB4181 when a task returns false (fails) without logging any errors

File: Directory.Build.props (hunk 2 above)
Comment (Pilchie) [reply Directory.Build.props:103]: I had looked at it before.  It turns out that we have *a lot* of debt there that would make it very hard to turn back on anytime soon.

File: src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj (hunk 1 above)
Comment (Pilchie) [reply src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj:17]: We probably *do* want the second one to warn.

**Review by Pilchie — APPROVED** [review]

**Comment by dougbu** [comment]
Internal build https://dev.azure.com/dnceng/internal/_build/results?buildId=694625 built all product code, including site extensions. That build did not compile any test code but PR validation build https://dev.azure.com/dnceng/public/_build/results?buildId=694621 covered that. Both hit only the expected (for now) 8 or 9 per-build non-`msbuild` warnings.

**Pull request merged**